src/main/java/com/openelements/conduct/service/AnalysisService.java

```diff
@@ -0,0 +1,178 @@
+                .collect(Collectors.groupingBy(
+                    report -> report.timestamp().format(DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+                ));
+
+        int averageNoViolationCountPerDayInThisWeek = calculateDailyAverage(thisWeekReportsByDay, ViolationState.NONE);
+        int averagePossibleViolationCountPerDayInThisWeek = calculateDailyAverage(thisWeekReportsByDay, ViolationState.POSSIBLE_VIOLATION);
+        int averageViolationCountPerDayInThisWeek = calculateDailyAverage(thisWeekReportsByDay, ViolationState.VIOLATION);
+
+        // Calculate growth metrics
+        LocalDateTime lastWeekStart = LocalDateTime.now().minusDays(14);
+        LocalDateTime lastWeekEnd = LocalDateTime.now().minusDays(7);
+        
+        List<ViolationReport> lastWeekReports = reports.stream()
+                .filter(r -> r.timestamp().isAfter(lastWeekStart) && r.timestamp().isBefore(lastWeekEnd))
+                .collect(Collectors.toList());
+
+        double generalGrowthOfChecksInPercentage = calculateGrowthPercentage(lastWeekReports.size(), thisWeekReports.size());
+        double growthOfNoViolationCountAgainstLastWeek = calculateGrowthPercentage(
+                (int) lastWeekReports.stream().filter(r -> r.violationState() == ViolationState.NONE).count(),
+                (int) thisWeekReports.stream().filter(r -> r.violationState() == ViolationState.NONE).count()
+        );
+        double growthOfPossibleViolationCountAgainstLastWeek = calculateGrowthPercentage(
+                (int) lastWeekReports.stream().filter(r -> r.violationState() == ViolationState.POSSIBLE_VIOLATION).count(),
+                (int) thisWeekReports.stream().filter(r -> r.violationState() == ViolationState.POSSIBLE_VIOLATION).count()
+        );
+        double growthOfViolationCountAgainstLastWeek = calculateGrowthPercentage(
+                (int) lastWeekReports.stream().filter(r -> r.violationState() == ViolationState.VIOLATION).count(),
+                (int) thisWeekReports.stream().filter(r -> r.violationState() == ViolationState.VIOLATION).count()
+        );
+
+        return new AnalysisDto(
+            totalNoViolationCount,
+            totalPossibleViolationCount,
+            totalViolationCount,
+            averageNoViolationCountPerDay,
+            averagePossibleViolationCountPerDay,
+            averageViolationCountPerDay,
+            maxNoViolationCountPerDay,
+            maxPossibleViolationCountPerDay,
+            maxViolationCountPerDay,
+            averageNoViolationCountPerDayInThisWeek,
+            averagePossibleViolationCountPerDayInThisWeek,
+            averageViolationCountPerDayInThisWeek,
+            generalGrowthOfChecksInPercentage,
+            growthOfNoViolationCountAgainstLastWeek,
+            growthOfPossibleViolationCountAgainstLastWeek,
+            growthOfViolationCountAgainstLastWeek,
+            LocalDateTime.now()
+        );
+    }
+
+    public TrendSummaryDto generateTrendSummary() {
```

Comment:
I like the AnalysisDto and assume that the containing numbers can be helpful. The TrendSummaryDto currently looks not that useful to me. I believe the best is to remove it for now and only use the AnalysisDto until we have a clear idea about trend analysis